contract/src/lib.rs

```diff
@@ -309,14 +309,10 @@ impl MpcContract {
             payload,
             path
         );
-        match self.pending_requests.get(&payload) {
-            None => {
-                self.pending_requests.insert(&payload, &None);
-                log!(&serde_json::to_string(&near_sdk::env::random_seed_array()).unwrap());
-                Self::ext(env::current_account_id()).sign_helper(payload, 0)
-            }
-            Some(_) => env::panic_str("Signature for this payload already requested"),
-        }
+        // TODO: prevent submitting the same paload twice which can cause conflicts
+        self.pending_requests.insert(&payload, &None);
+        log!(&serde_json::to_string(&near_sdk::env::random_seed_array()).unwrap());
+        Self::ext(env::current_account_id()).sign_helper(payload, 0)
```

Comment:
Let's split this off into several cases since we don't want to potentially erase the signature if it's already there:
```suggestion
        match self.pending_requests.get(&payload) {
            None => {
                self.pending_requests.insert(&payload, &None);
                log!(&serde_json::to_string(&near_sdk::env::random_seed_array()).unwrap());
                Self::ext(env::current_account_id()).sign_helper(payload, 0)
            }
            Some(None) => Self::ext(env::current_account_id()).sign_helper(payload, 0),
            Some(Some(signature)) => PromiseOrValue::Value(signature),
        }
```

Reply:
Additionally, let's just remove the removal of the pending signatures, such that this contract just caches the signatures for now

Reply:
Resolved with the commit I pushed